### ISSUE TITLE
actionbar support

### ISSUE THREAD

**Issue by maks** [Jasonette/JASONETTE-Android#122]
In Android, the "header" idiomatic UI is an actionbar with an options items menu [as per](https://developer.android.com/guide/topics/ui/menus.html#options-menu)

I think this also needs an issue in https://github.com/Jasonette/documentation as currently the json spec only supports iOS style single action "menu icon" on right-hand side of the "header".

The existing code already uses a support lib toolbar and does `setSupportActionBar(toolbar)` so its about half way there already.

**Comment by gliechtenstein** [comment]
There's a related PR on the iOS side https://github.com/Jasonette/JASONETTE-iOS/pull/221

Basically we're thinking of implementing the left button as well, but I don't think we can implement ALL the features the action bar includes as that will break the cross-platform nature of the framework. For example, on Android you can add multiple buttons on the right side but I think we should only allow one on the right, and one on the left.

**Comment by maks** [comment]
Great!
Not sure what you mean by "multiple buttons"?
On Android the actionbar gets given a list of items that it needs to display, it decides (with optional hints from app code) which are shown in the appbar as icons and which can't fit and go into the overflow drop-down menu  (ie. the 3 vertical stacked dots).
I'm not very familiar with iOS UI but I dont think there is a comparable UI component?

I think there maybe enough differences between the 2 platforms that the json spec could have elements that are only applicable on one platform or the other, otherwise using jasonette means limiting yourself to just the UI elements common to both platforms

**Comment by gliechtenstein** [comment]
> I think there maybe enough differences between the 2 platforms that the json spec could have elements that are only applicable on one platform or the other, otherwise using jasonette means limiting yourself to just the UI elements common to both platforms

I think you could say the same thing about web browsers. You can't use all the cool features of Mac on a web browser, and same for Windows. 

But the benefit of using a web browser is that they are consistent across platforms, which is a huge benefit since you only need to write once to run everywhere. The top priority of Jasonette is to simplify and make it easy to build apps, and making the core support different sets of markup for different platforms goes against this goal.

That said, the plan is to: 

1. Core: only support the common parts, so that it's cross platform and easy to build.
2. Extension: For those who want more customization, provide easy ways to do so via extensions.

Currently Jasonette is focused on step 1, but we'll get to step 2 in the future. I think it's important to make sure that the core is cross platform. Hope this makes sense.

**Comment by maks** [comment]
Yes thanks, that does make very good sense. In regards to extensions, I'm guessing the same approach can be taken for Android as for iOS thats [documented here](https://docs.jasonette.com/advanced/)?
I guess that's what https://github.com/Jasonette/documentation/issues/36 is tracking ?

**Comment by gliechtenstein** [comment]
@maks yes, you're right. 

But we also have made quite a progress since the documentation was written (although nothing has been officially released yet) 

Basically, while it's easy to write extensions yourself, it's very hard to share and reuse someone else's extensions right now because there's no uniform way of distributing and integrating into existing Jasonette projects. So a couple of us worked on [Jason Extension Manager](https://github.com/Jasonette/JEM) (related thread [here](https://github.com/Jasonette/JASONETTE-iOS/issues/107)), it's 90% finished and I was planning to release it officially, but due to a sudden priority change I had to shelf it for a while. I plan on getting back to it full time once I finish the current priority task I'm working on (web), but the extension manager itself is pretty functional so feel free to play with it!

**Comment by realitix** [comment]
My 2 cents: Agreed with @gliechtenstein : Support one left + one right button (like Spotify app).
